src/app/components/reports/pages/event-summary/new/new.component.ts

```diff
@@ -0,0 +1,289 @@
+  regions: BehaviorSubject<Region[]>;
+  sites: BehaviorSubject<Site[]>;
+  provenances: BehaviorSubject<AudioEventProvenance[]>;
+  provenanceScoreCutOff: number;
+  charts: BehaviorSubject<string[]>;
+  eventsOfInterest: BehaviorSubject<Tag[]>;
+  binSize: BinSize;
+}
+
+@Component({
+  selector: "baw-new-summary-report",
+  templateUrl: "./new.component.html",
+  styleUrls: ["./new.component.scss"],
+})
+class NewEventReportComponent extends PageComponent implements OnInit {
+  public constructor(
+    protected sitesApi: SitesService,
+    protected regionsApi: RegionsService,
+    protected provenanceApi: AudioEventProvenanceService,
+    protected tagsApi: TagsService,
+    private route: ActivatedRoute,
+    private router: Router
+  ) {
+    super();
+  }
+
+  public project: Project;
+  public region?: Region;
+  public site?: Site;
+
+  public model: IEventReportConditions = {
+    dateTime: new BehaviorSubject<AudioRecordingFilterModel>({}),
+    regions: new BehaviorSubject<Region[]>([]),
+    sites: new BehaviorSubject<Site[]>([]),
+    provenances: new BehaviorSubject<AudioEventProvenance[]>([]),
+    provenanceScoreCutOff: 0.8,
+    charts: new BehaviorSubject<string[]>([]),
+    eventsOfInterest: new BehaviorSubject<Tag[]>([]),
+    binSize: BinSize.month,
+  };
+
+  public ngOnInit(): void {
+    const models = retrieveResolvers(this.route.snapshot.data as IPageInfo);
+
+    if (!hasResolvedSuccessfully(models)) {
+      return;
+    }
+
+    if (models[projectKey]) {
+      this.project = models[projectKey] as Project;
+    }
+
+    if (models[regionKey]) {
+      this.region = models[regionKey] as Region;
+    }
+
+    if (models[siteKey]) {
+      this.site = models[siteKey] as Site;
+    }
+  }
+
+  public generateReport(): void {
+    if (!this.isValidProvenanceCutOff()) {
+      return;
+    }
+
+    const parameterModel = this.createSearchParametersDataModel();
+    const queryStringParameters = parameterModel.toQueryString();
+
+    this.router.navigateByUrl(
+      this.viewReportUrl() + `?${queryStringParameters}`
+    );
+  }
+
+  // while this could be generalised, I have chosen not to as it'd lose a lot of strong type safety without the use of generics
+  // as we are using these formatters in ng html templates, it is not possible to use TypeScript generics to enfore type safety
+  // on a generalised function
+  public regionFormatter = (region: Region) => region.name;
+  public siteFormatter = (site: Site) => site.name;
+  public provenanceFormatter = (provenance: AudioEventProvenance) =>
+    provenance.name;
+  public chartsFormatter = (chart: string) => chart;
+  public eventsOfInterestFormatter = (tag: Tag): string => tag.text;
+
+  public regionsOptions = (regionName: string): Observable<Region[]> => {
+    const filter: Filters<IRegion> = {
+      filter: {
+        name: {
+          contains: regionName,
+        },
+      },
+    };
+
+    return this.regionsApi.filter(filter, this.project);
```

Comment:
```suggestion
    return this.regionsApi.filter(this.regionsApi.nameFilter(regionName), this.project);
```
```suggestion
    return this.shallowRegionsApi.filter(filters.and(NameFilter(regionName), ProjectFilter(this.project)));
```

you should be able to make a `NameFilter` function that works on any model with a `name` property and is strongly typed

Reply:
100% agree

I'm probably going to change the function name to something like "`propertyFilter`" instead as _technically_, we're not filtering by name, we're filtering by a key-value property on an object.

E.g. `Tag` models don't have names, but they do have `text` that we filter for in the generation page

So the signature of the function will be something like

```ts
function propertyFilter<T>(key: string keyof T, value: forgotSyntax)
```